cinema/tests/test_movie_viewset.py

```
@@ -0,0 +1,96 @@
+from io import BytesIO
+from PIL import Image
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework.test import APIClient
+from rest_framework import status
+from cinema.models import Movie, Genre, Actor
+from django.contrib.auth import get_user_model
+from django.core.files.uploadedfile import SimpleUploadedFile
+
+User = get_user_model()
+
+
+def get_test_image():
+    file = BytesIO()
+    image = Image.new("RGB", (100, 100), color="red")
+    image.save(file, "JPEG")
+    file.seek(0)
+    return SimpleUploadedFile(
+        "test.jpg", file.read(), content_type="image/jpeg"
+    )
+
+class MovieViewSetTestCase(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(
+            email="admin.user@cinema.com", password="1qazcde3", is_staff=True
+        )
+        self.client = APIClient()
+        token_url = reverse("token_obtain_pair")
+        response = self.client.post(
+            token_url,
+            {"email": self.user.email, "password": "1qazcde3"},
+            format="json"
+        )
+        self.token = response.data["access"]
+        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer {self.token}")
```

After posting to the token endpoint you immediately use response.data['access'] without asserting the token request succeeded. Add an assertion to make failures clear, for example: assert response.status_code == 200 and 'access' in response.data before using the token. This will make setUp fail fast if token obtaining breaks.